src/core/document.js

```diff
@@ -826,6 +827,53 @@ class PDFDocument {
     return this.xfaFactory !== null;
   }
 
+  loadXfaFonts(handler) {
+    const resources = this.catalog.acroForm.get("DR");
+    const fonts = resources.get("Font");
+    if (!(fonts instanceof Dict)) {
+      return Promise.resolve(null);
+    }
+
+    const objectLoader = new ObjectLoader(resources, ["Font"], this.xref);
+    const partialEvaluator = new PartialEvaluator({
+      xref: this.xref,
+      handler,
+      pageIndex: 0,
+      idFactory: this._globalIdFactory,
+      fontCache: this.catalog.fontCache,
+      builtInCMapCache: this.catalog.builtInCMapCache,
+    });
+    const operatorList = new OperatorList();
+    const initialState = {
+      font: null,
+      clone() {
+        return this;
+      },
+    };
+
+    return objectLoader.load().then(() => {
+      fonts.forEach((pdfFontName, font) => {
+        const descriptor = font.get("FontDescriptor");
+        if (descriptor instanceof Dict) {
+          const fontFamily = descriptor.get("FontFamily");
+          const fontWeight = descriptor.get("FontWeight");
+          const italicAngle = descriptor.get("ItalicAngle");
+
+          partialEvaluator.handleSetFont(
+            resources,
+            [Name.get(pdfFontName), 1],
+            null,
+            operatorList,
+            "",
```

Comment:
This doesn't really seem like a good idea, since leaving the `task` both empty and of the wrong type could (with future changes) lead to errors in the `evaluator.js`-code.

At the very least, I think that you should instead construct a "dummy"-task with all of the expected methods/properties (although they can of course be empty).